tfrecord_loader.py

```diff
@@ -11,6 +11,10 @@ def __init__(self, index_fname, batch_size, parse_fn, map_fn=None, restore_state
             self.file_idx = restore_state["file_idx"]
             self.file_idx_init = False
             self.used = restore_state["used"]
+
+            unique = set(self.used)
+            last_pass_start_idx = len(self.used) % len(unique)
+            self.used = self.used[-last_pass_start_idx:]
```

Comment:
I'm not sure why this is nessasary, can you reset self.used around the `except StopIteration:` line?

Reply:
Oh, yeah, that's a better idea

Changed it to just do `self.reset()` in that except block